test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -397,13 +398,18 @@ class test_uct_cm_sockaddr : public uct_test {
 
         self->m_cm_state |= TEST_CM_STATE_CONNECT_REQUESTED;
 
-        if (!self->m_reject_conn_request) {
-            self->m_server->accept(conn_request, server_connect_cb,
-                                   server_disconnect_cb, self);
-        } else {
+        if (self->m_switch_cm) {
```

Comment:
maybe add enum with test mode to unite m_switch_cm, m_reject_conn_request and default?

Reply:
i think we can inherit a new test class, which has m_cm_test field. there will be a virtual function accept() which will use the default cm in the base class, and m_test_cm in the new class,
m_cm_test could be created in ::init() of the new class